testing/integration/src/integration_tests.rs

```diff
@@ -1703,3 +1709,77 @@ async fn staging_consensus_test() {
     core.shutdown();
     core.join(joins);
 }
+
+#[tokio::test]
+async fn sanity_integration_test() {
+    let core1 = DaemonWithRpc::new_random();
+    let (workers1, rpc_client1) = core1.start().await;
+
+    let core2 = DaemonWithRpc::new_random();
+    let (workers2, rpc_client2) = core2.start().await;
+
+    tokio::time::sleep(Duration::from_secs(1)).await;
+    rpc_client1.disconnect().await.unwrap();
+    drop(rpc_client1);
+    core1.core.shutdown();
+    core1.core.join(workers1);
+
+    rpc_client2.disconnect().await.unwrap();
+    drop(rpc_client2);
+    core2.core.shutdown();
+    core2.core.join(workers2);
+}
+
+struct DaemonWithRpc {
+    core: Arc<Core>,
+    rpc_port: u16,
+}
+
+impl DaemonWithRpc {
+    fn new_random() -> DaemonWithRpc {
+        let mut args = Args { devnet: true, ..Default::default() };
+
+        // This should ask the OS to allocate free port for socket 1 to 4.
+        let socket1 = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
+        let rpc_port = socket1.local_addr().unwrap().port();
+
+        let socket2 = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
+        let p2p_port = socket2.local_addr().unwrap().port();
+
+        let socket3 = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
+        let rpc_json_port = socket3.local_addr().unwrap().port();
+
+        let socket4 = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
+        let rpc_borsh_port = socket4.local_addr().unwrap().port();
+
+        drop(socket1);
+        drop(socket2);
+        drop(socket3);
+        drop(socket4);
+
+        args.rpclisten = Some(format!("0.0.0.0:{rpc_port}").try_into().unwrap());
+        args.listen = Some(format!("0.0.0.0:{p2p_port}").try_into().unwrap());
+        args.rpclisten_json = Some(format!("0.0.0.0:{rpc_json_port}").parse().unwrap());
+        args.rpclisten_borsh = Some(format!("0.0.0.0:{rpc_borsh_port}").parse().unwrap());
+        args.appdir = Some(tempdir().unwrap().path().to_str().unwrap().to_owned());
```

Comment:
I think we need to store `tempdir` within `DaemonWithRpc ` so that it is dropped only at the end

Reply:
from the tempdir crate docs:


[Resource Leaking](https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#resource-leaking)
Various platform-specific conditions may cause TempDir to fail to delete the underlying directory. It’s important to ensure that handles (like [File](http://doc.rust-lang.org/std/fs/struct.File.html) and [ReadDir](http://doc.rust-lang.org/std/fs/struct.ReadDir.html)) to files inside the directory are dropped before the TempDir goes out of scope. The TempDir destructor will silently ignore any errors in deleting the directory; to instead handle errors call [TempDir::close()](https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#method.close).